create nav bar styling and routes

AnshaalHussain opened the pull request:
## Description

Created navbar that redirects to website url paths such as _map_, _home_, and _about us_

## Tasks

- created flask routes in ```__init__.py```
- added navbar in ```index.html``` template
- added navbar styling in ```main.css```
- added pages data list in ```app/data.json```

## Resources and screenshots (optional)

https://hackersandslackers.com/flask-routes/

AnshaalHussain committed:
create nav bar styling and routes

dmosc commented on app/__init__.py:
```diff
@@ -11,7 +11,17 @@
+        return render_template('index.html', title="Week 1 - Team Portfolio", url=os.getenv("URL"), users=data["users"], pages=data["pages"])
+
+
+@app.route('/map')
+def map():
+    return "MAP SHOULD BE HERE"
+
+
+@app.route('/aboutme')
+def aboutme():
+    return "ABOUT ME"
```
I think we might leave this off to the pr that actually implements them. It's a good idea to ship code that does a task. Maybe leave TODO comments?

```
// TODO: Implement /map route.
// TODO: Implement /aboutme route.
```

AnshaalHussain replied:
Makes sense, I'll replace them with TODO comments. I had made them to make sure routing functionality works with the nav bar links.

dmosc commented on app/data.json:
```diff
@@ -62,5 +62,8 @@
       ]
     }
-  ]
+  ],
+  
+  "pages": [{"name": "Home", "url": "http://127.0.0.1:5000/"}, {"name": "Map", "url": "http://127.0.0.1:5000/map"}, {"name": "About Us", "url": "http://127.0.0.1:5000/aboutme"}]
```
I first intended to use this file to "mirror" a database that fetches information related to our users. The "configuration" parameter you're suggesting would fit better in a file like: `app/config.json` (which is something I'm totally down for).

Else, I think hard-coding these values inside the HTML sounds like a straightforward solution as well.

LauraAurora replied:
I'm open to both solutions.

AnshaalHussain replied:
I'm interested in the ```app/config.json``` idea! I need some more clarification on what that would look like however.

AnshaalHussain replied:
I've hardcoded the URLs values for now.

dmosc commented on app/static/styles/main.css:
```diff
@@ -31,6 +31,22 @@ body {
   padding-top: calc(var(--padding) * 2);
 }
 
+.navbar-section li a {
+  font-weight: 500;
+  font-size: 16px;
+  text-decoration: none;
+
+  color: var(--lightgrey);
+
+}
+
+.navbar-content {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  padding: 30px 4%;
+}
+
```
All this is related to the previous comment on implementing a feature altogether.

AnshaalHussain replied:
Sorry, I'm not sure what you mean by this comment.

dmosc replied:
Oh actually nvm, these are rules for the navbar; my bad.

dmosc commented on app/static/styles/main.css:
```diff
@@ -41,6 +57,23 @@ body {
   height: 60px;
 }
 
+.nav__links {
+  list-style: none;
+}
+
+.nav__links li {
+  display: inline-block;
+  padding: 0px 20px;
+}
+
+
+.nav__links li a:hover {
+
+color: var(--yellow)
+}
+
+
+
```
On another note, I think we should stick to a uniform naming convention for CSS. All existing styles use `dash-names` :))

dmosc commented on app/templates/index.html:
```diff
@@ -23,15 +23,24 @@
             <img width="200px" src="/static/images/mlh-logo.svg" />
           </div>
         </a>
+        <nav>
+          <ul class="nav__links">
+            {% for page in pages %}
+            <li><a href="{{ page.url }}">{{ page.name }}</a></li>
+            {% endfor %}
+          </ul>
+        </nav>
```
^

dmosc reviewed (changes requested):
Hi, thanks for the work!

Left some comments. Also, I simultaneously opened the [3d rendering pr](https://github.com/MLH-Fellowship/project-placeholder/pull/21) and some files are going to conflict on the pr that gets merged second. I think we should pick one to be merged first and the other will have to resolve conflicts.

Let me know what sounds best for you.

AnshaalHussain committed:
maintain consistent css naming conventions

AnshaalHussain committed:
maintain consistent css class naming conventions

update index.html to reflect update to css class naming of ```nav__links```

AnshaalHussain committed:
Update __init__.py

remove routes that currently have no functionality, and leave TODO comments for implementing later

AnshaalHussain committed:
Update index.html

Changed nav bar links to now be hard coded with URLs

AnshaalHussain committed:
Update __init__.py

removed "pages=data["pages"]" that was being passed into the index.html

AnshaalHussain committed:
Update data.json

remove pages data array from app/data.json

dmosc commented on app/static/styles/main.css:
```diff
@@ -41,6 +57,23 @@ body {
+color: var(--yellow)
+}
+
+
+
```
rm

dmosc reviewed (approved):
Nice; thank you so much for the effort. It looks awesome.

Just have to resolve the conflicts. If you think it might help, we could definitely set up a pair programming session and I can help you out.

AnshaalHussain committed:
Merge branch 'main' into 11-add-a-menu-bar-that-dynamically-displays-other-pages-in-the-app
